### PR TITLE
dvtm 0.13

### PR DESCRIPTION
**Pull request by afh**

### CODE REVIEW

File: Library/Formula/dvtm.rb [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
+  url "http://www.brain-dump.org/projects/dvtm/dvtm-0.13.tar.gz"
+  sha1 "8a4fc2440faa3050244e5a492fb6766899e0c0d7"
   head "git://repo.or.cz/dvtm.git"
 
   def install
-    inreplace "config.mk", "LIBS = -lc -lutil -lncursesw", "LIBS = -lc -lutil -lncurses"
-    inreplace "Makefile", "strip -s", "strip"
-    system "make", "PREFIX=#{prefix}", "install"
+    ENV.append_to_cflags "-D_DARWIN_C_SOURCE"
+    system "make", "PREFIX=#{prefix}", "LIBS=-lc -lutil -lncurses", "install"
```
Comment (mikemcquaid) [inline Library/Formula/dvtm.rb]: Any chance of a wee `test do` for this?

File: Library/Formula/dvtm.rb [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+  end
+
+  test do
+    assert_match /^dvtm-[0-9.]+ © 2007-\d{4} Marc André Tanner$/, shell_output("#{bin}/dvtm -v")
```
Comment (afh) [inline Library/Formula/dvtm.rb]: `dvtm` is a virtual terminal manager, an interactive program similar to `tmux` and `screen` so this is the best test I could come up with right now.

Comment (mikemcquaid) [inline Library/Formula/dvtm.rb]: Thanks!

**Comment by tdsmith** [comment]
Test is failing:

> Testing dvtm
==> /usr/local/Cellar/dvtm/0.13/bin/dvtm -v
Error: dvtm: failed
invalid byte sequence in US-ASCII
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/minitest/unit.rb:302:in `=~'

Does the formula need a [special encoding comment](http://ruby-doc.org/core-2.2.0/Encoding.html#class-Encoding-label-Script+encoding)?

**Comment by afh** [comment]
@tdsmith Yes it does need a special encoding comment, thanks!
I've updated the formula accordingly.

**Comment by afh** [comment]
It seems the special encoding comment did not fix the issue. Any other ideas?

File: Library/Formula/dvtm.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/dvtm.rb]: Just do a partial match that doesn't require the unicode.

**Comment by afh** [comment]
This is getting ridiculous :smile: I've checked that the version string does not contain any special unicode whitespace characters (all are 0x20) I'm out of ideas and open for suggestions.

File: Library/Formula/dvtm.rb [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
   def install
-    inreplace "config.mk", "LIBS = -lc -lutil -lncursesw", "LIBS = -lc -lutil -lncurses"
-    inreplace "Makefile", "strip -s", "strip"
-    system "make", "PREFIX=#{prefix}", "install"
+    ENV.append_to_cflags "-D_DARWIN_C_SOURCE"
+    system "make", "PREFIX=#{prefix}", "LIBS=-lc -lutil -lncurses", "install"
+  end
+
+  test do
+    assert_match /^dvtm-[0-9.]+ . 2007-\d{4} Marc Andr. Tanner$/, shell_output("#{bin}/dvtm -v")
```
Comment (mikemcquaid) [inline Library/Formula/dvtm.rb]: Just go for `system "dvtm", "-v"`

**Comment by jacknagel** [comment]
Can reproduce by running `LC_ALL=C brew test dvtm`. I think this will make it pass:

```diff
diff --git a/Library/Formula/dvtm.rb b/Library/Formula/dvtm.rb
index 9234791..3672ef1 100644
--- a/Library/Formula/dvtm.rb
+++ b/Library/Formula/dvtm.rb
@@ -1,4 +1,3 @@
-# encoding: UTF-8
 class Dvtm < Formula
   homepage "http://www.brain-dump.org/projects/dvtm/"
   url "http://www.brain-dump.org/projects/dvtm/dvtm-0.13.tar.gz"
@@ -11,6 +10,8 @@ def install
   end
 
   test do
-    assert_match /^dvtm-[0-9.]+ . 2007-\d{4} Marc Andr. Tanner$/, `#{bin}/dvtm -v`
+    out = shell_output("#{bin}/dvtm -v")
+    out.force_encoding("UTF-8") if out.respond_to?(:force_encoding)
+    assert_match /^dvtm-[0-9.]+ . 2007-\d{4} Marc Andr. Tanner$/, out
   end
 end
```

Externally generated strings are normally tagged with the environment encoding, which is C / US-ASCII on the bots.

**Comment by afh** [comment]
@jacknagel Thanks for your comments.

**Pull request closed without merging**